### PR TITLE
Fix delete scene request + mock data issues on object level

### PR DESCRIPTION
**Pull request by ahmedhamodi**
When a scene was deleted it was not also removed from the scene_ids array on the scenario level. So when a subsequent fetch was made for the scenes, we would get an error. Also, the mock data for objects was not properly setup since we do not allow same objects in multiple rooms, so when a scene is deleted so are all the nested objects, but then that causes issues for the scene since its objects are deleted. Made them independent on mock data.

### CODE REVIEW

File: scripts/mockdata/scenes.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   {
     "name": "Living Room",
     "description": "A standard master bedroom. King size bed, a dresser, ensuit washroom, and a large closet.",
-    "object_ids": [2],
+    "object_ids": [12],
```
Comment (jaydhulia) [inline scripts/mockdata/scenes.json:29]: This should still be 2, 2 is the sanitizer in the Living room.

File: scripts/mockdata/objects.json [hunk 2]
```diff
@@ -301,5 +301,143 @@
...
+        "jsonData": {
+          "password": "abcde",
+          "model": "basic",
+          "is_last_object": true
+        }
+      }
+    }
+  },
+  {
+    "name": "Classroom keypad",
```
Comment (jaydhulia) [inline scripts/mockdata/objects.json:402]: This object isn't needed

File: scripts/mockdata/objects.json [hunk 3]
```diff
@@ -301,5 +301,143 @@
...
+        "jsonData": {
+          "password": "1234",
+          "model": "numpad",
+          "is_last_object": true
+        }
+      }
+    }
+  },
+  {
+    "name": "Classroom keyboard",
```
Comment (jaydhulia) [inline scripts/mockdata/objects.json:374]: Can you rename the objects to map appropriately? For example, this is the Student bedroom keyboard, not the classroom keyboard.

File: app/library/postgres.py [hunk 4]
```diff
@@ -209,6 +209,16 @@ async def delete_scene_from_postgres(scene_id: str, session):
     if not delete_scene(scene_id, session):
         raise ValueError("Scene ID not valid")
 
+    # Remove the scene from scenario's scene_ids array
+    scenarios = get_scenarios(session)
+    for scenario in scenarios:
+        if int(scene_id) in scenario.scene_ids:
+            scenario_obj: Scenario = await get_scenario_from_postgres(
```
Comment (jaydhulia) [inline app/library/postgres.py:216]: I am confused... why are you getting the `scenario_obj` here? You already have the scenario (`for scenario in scenarios:`), why is there a need to make a second request to postgres to get the same scenario?

File: app/library/postgres.py [hunk 5]
```diff
@@ -209,6 +209,16 @@ async def delete_scene_from_postgres(scene_id: str, session):
...
+                scenario.id, session
+            )
+            scenario_obj["scene_ids"].remove(int(scene_id))
```
Comment (jaydhulia) [inline app/library/postgres.py:219]: `scenario_obj` is not a dict, on line `scenario_obj: Scenario `. So you should use `scenario_obj.scene_ids` if it is a scenario, OR change the above line to  `scenario_dict: dict`. But like the previous comment mentions, scenario_obj is redundant.  
image

**Review by jaydhulia — CHANGES_REQUESTED** [review]

File: app/library/postgres.py (hunk 4 above)
Comment (ahmedhamodi) [reply app/library/postgres.py:216]: Because I am being silly. Not needed!

**Comment by aaronabraham311** [comment]
@ahmedhamodi I used the exact same logic that Kevin used [here](https://github.com/uwblueprint/cdc-frontend/blob/69ab646609d3a3a52ae942973617885e37cef58e/src/Components/Admin/EnvironmentEditor/environment.js#L135) for scene creation. Is it worth changing the scene creation endpoint to automatically update the scenario's `scene_ids` field as well?

**Comment by jaydhulia** [comment]
@aaronabraham311 I don't think we should do that. Scenes are independent of the scenario, and scene creation shouldn't automatically add to any scenario. It makes sense for deletion, but not addition imo.

**Review by jaydhulia — APPROVED** [review]

**Pull request merged**